crates/rust-analyzer/src/config.rs

```diff
@@ -827,16 +1177,45 @@ impl Config {
         caps: ClientCapabilities,
         workspace_roots: Vec<AbsPathBuf>,
         visual_studio_code_version: Option<Version>,
+        user_config_path: Option<Utf8PathBuf>,
```

Comment:
question: are there non-`user` config paths? i would just rename all usage here to `config_path`, IMO.

(i see a "root config path", but I'd remove the `user` qualifier entirely.)

Reply:
Yeah naming things is not my strength. I though of XDG_CONFIG_.. but that is also not very accurate either. Matter of fact it was Veykril's suggestion to change it to user_config. In any case the name is still the same but I added comment.